sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/WriteToDataSourceV2Exec.scala

```diff
@@ -612,6 +616,12 @@ case class DeltaWithMetadataWritingSparkTask(
 private[v2] trait V2CreateTableAsSelectBaseExec extends LeafV2CommandExec {
   override def output: Seq[Attribute] = Nil
 
```

Comment:
I checked and all sub-classes of `V2CreateTableAsSelectBaseExec` has a catalog field, so we can add
```
def catalog: TableCatalog
```
in this base class and implement `val metrics: Map[String, SQLMetric]` here.